docs/guides/vault-as-secret-store.md

```diff
@@ -0,0 +1,452 @@
+for the cluster where Crossplane is running.
+
+### Prepare Vault
+
+1. Install Vault Helm Chart
+
+```shell
+kubectl create ns vault-system
+
+helm repo add hashicorp https://helm.releases.hashicorp.com --force-update
+helm -n vault-system upgrade --install vault hashicorp/vault
+```
+
+2. [Unseal] Vault
+
+```shell
+kubectl -n vault-system exec vault-0 -- vault operator init -key-shares=1 -key-threshold=1 -format=json > cluster-keys.json
+VAULT_UNSEAL_KEY=$(cat cluster-keys.json | jq -r ".unseal_keys_b64[]")
+kubectl -n vault-system exec vault-0 -- vault operator unseal $VAULT_UNSEAL_KEY
+```
+
+3. Configure Vault with Kubernetes Auth.
+
+In order for Vault to be able to authenticate requests based on Kubernetes 
+service accounts, the [Kubernetes auth method] must be enabled.
+This requires logging in to Vault and configuring it with a service account
+token, API server address, and certificate. Because we are running Vault in
+Kubernetes, these values are already available via the container filesystem and
+environment variables.
+
+```shell
+cat cluster-keys.json | jq -r ".root_token" # get root token
+
+kubectl -n vault-system exec -it vault-0 -- /bin/sh
+vault login # use root token from above
+vault auth enable kubernetes
+
+vault write auth/kubernetes/config \
+        token_reviewer_jwt="$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" \
+        kubernetes_host="https://$KUBERNETES_PORT_443_TCP_ADDR:443" \
+        kubernetes_ca_cert=@/var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+        
+exit # exit vault container
+```
+
+4. Enable Vault Key Value Secret Engine
+
+There are two different versions of [Vault KV Secrets Engine], `v1` and `v2`,
+which you can find more details in the linked documentation page. 
+We will use `v2` in this guide as an example, however, both versions are
+supported as an external secret store.
+
+```shell
+kubectl -n vault-system exec -it vault-0 -- vault secrets enable -path=secret kv-v2
+```
+
+5. Create a Vault Policy and Role for Crossplane
+
+```shell
+kubectl -n vault-system exec -i vault-0 -- vault policy write crossplane - <<EOF
+path "secret/data/*" {
+    capabilities = ["create", "read", "update", "delete"]
+}
+path "secret/metadata/*" {
+    capabilities = ["create", "read", "update", "delete"]
+}
+EOF
+
+kubectl -n vault-system exec -it vault-0 -- vault write auth/kubernetes/role/crossplane \
+    bound_service_account_names="*" \
+    bound_service_account_namespaces=crossplane-system \
+    policies=crossplane \
+    ttl=24h
+```
+
+### Install and Configure Crossplane
+
+1. Install Crossplane by:
+
+- Enabling `External Secret Stores` feature. 
+- Annotating for [Vault Agent Sidecar Injection]
+
+```shell
+kubectl create ns crossplane-system
+
+helm repo add crossplane-stable https://charts.crossplane.io/stable --force-update
+
+helm --namespace crossplane-system upgrade --install crossplane crossplane-stable/crossplane --namespace crossplane-system \
```

Comment:
Should we provide a `--version` flag to this command? I tried with the `--version` unspecified and it caused an `Init:Error` in crossplane Pod startup because the default version used was `v1.2` and the flag `--enable-external-secret-stores` isn't available.

If we are already using the provider-gcp image `turkenh/provider-gcp:v0.21.0-rc.13.g273ebf2` in this guide, can't we set the `--version` to something like `v1.8.0-rc.0.5.ge78e6a7a` (the latest today)?

Reply:
I guess you didn't update your local helm repo cache, because I would expect you to have the latest v1.6 release. I was expecting the following will do the trick, did you run that command as well? (or skipped thinking you already have it added):

```
helm repo add crossplane-stable https://charts.crossplane.io/stable --force-update
```

Reply:
> If we are already using the provider-gcp image turkenh/provider-gcp:v0.21.0-rc.13.g273ebf2 in this guide, can't we set the --version to something like v1.8.0-rc.0.5.ge78e6a7a (the latest today)?

Honestly, I am not thinking to merge this PR until we have the release. So, I will update those images before merging.